### PR TITLE
Feature/add factory links prop

### PR DESCRIPTION
**Pull request by miduga**
* New optional prop factoryLink, used to create the link inside the pagination-item. By default, it creates the link as before.
* Improved docs testing the new factoryLink prop with a <Link> React Router element.
* Move Pagination to a stateless component.

### CODE REVIEW

**Comment by kikoruiz** [comment]
👍 Arrebatando los estados allá donde va :hocho::european_castle:

**Comment by FranDelaCasa** [comment]
👍

File: src/sui-pagination-item/index.jsx [hunk 1]
```diff
@@ -22,32 +22,25 @@ export default function PaginationItem (props) {
...
+      { props.isCurrent
+        ? getText()
+        : props.factoryLink(linkProps) }
     </li>
   );
 }
 
 PaginationItem.propTypes = {
-  handlePaginate: PropTypes.func.isRequired,
+  factoryLink: PropTypes.func.isRequired,
```
Comment (ruben-martin-lozano) [inline src/sui-pagination-item/index.jsx]: It is not required right? :)

**Comment by ruben-martin-lozano** [comment]
A parte de lo del required para el factory, 👍

**Pull request merged**